### DISCUSSION TITLE
Disable component on multiple conditions

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#7916] [Help]
Hello,

We want to disable a component based on record status and user role ( can be custom role)

Example:  Record Status is Completed  &  User role is  Basic
![image](https://user-images.githubusercontent.com/42922833/191798725-4ff1e95c-97f9-4371-a822-76db22f5991d.png)

this setup is not working, component is not disabled when both condition are satisfied.

is there any other way we can achieve this?  may be through javascript or handlerbar conditions ? 

Thanks

**Comment by Zero2XIII** [comment]
Hi,

I have found in these cases to state the negative, there isn't a "AND" operator as far as I am aware.
In the above, I would do:
Set the field to be disabled in default.
1. Update setting away from disabled if user role is NOT EQUAL to basic
2. Update setting away from disabled if repeater... is NOT EQUAL to Complete

See if that maybe works for your case scenario?

A workaround that I have found for AND cases is to do something like the following (Basically concatenate the values to compare):
IF **{{Field One}} {{User Role ID}}** Equals **{{Value 1}} {{ Basic Role ID}}**

Hope it helps, cheers.

**Comment by kasgithub8** [comment]
Hi @applicationuser,

you could do it with JavaScript, the first IF would be a simple `if this equals this and this equals this then return this` and then you put whatever is being returned in the last field of your condition.

**Reply by applicationuser** [reply]
Hi @kasgithub8 ,

this is what i had in javascript, (may require some tweaks),  getting error- Error while executing JS
`if ($("Current User.roleId") == $("trim "BASIC"") and $("Repeater.Table.STATUS") == "Completed")
	return false;
else
	return true;`

**Reply by kasgithub8** [reply]
Well @applicationuser, you gotta bind in your Data Provider. If your Data Provider is called "Data Provider", you can just do it like this:

You put the condition on your component:
<img width="641" alt="image" src="https://user-images.githubusercontent.com/95284278/192965848-0766e087-f395-4221-80ea-754045d06de2.png">

With this JavaScript:
```
const json = $("Data Provider.Rows");
if(json[0]['Status'] == "Completed" && $("Current User.roleId") == "BASIC")
	{
		return "Show"
	} else {
		return "DontShow"
	}
```

Watch out for column names but I think I already used yours here.
